doc/MAPGEN.md

```diff
@@ -233,241 +249,307 @@ Example:
 
-Example: "x2": 22, "y2": [ 15, 20 ]
+See terrain.json, furniture.json, and trap.json for "id" strings.
 
-#### 2.2.1.2 "id"
-**required except by "radiation"** See terrain.json, furniture.json, and trap.json for "id" strings
-> Value: "ter_id", "furn_id", or "trap_id"
+### "point"
 
-Example: "id": "f_counter"
+| Field | Description
+|---    |---
+| "point" | (required) Set things by point. Value: `"terrain", "furniture", "trap", "radiation"`. Requires "x", "y", "id" (or "amount" for "radiation")
+| "id" | (required except by "radiation") Value: `"ter_id", "furn_id", or "trap_id"`. Example: `"id": "tr_beartrap"`
+| "x", "y" | (both required) X, Y coordinates, either as a constant from `0-23`, or as a range `[ 0-23, 0-23 ]` for a random point in that range. Example: `"x": 12, "y": [ 5, 15 ]`
+| "amount" | (required for "radiation") Value: 0-100. Radiation amount
+| "chance" | (optional) One-in-N chance to apply
+| "repeat" | (optional) Value: `[ from_num, to_num ]`. Repeat this randomly between `from_num` and `to_num` times. Only makes sense if the coordinates are random. Example: `[ 1, 3 ]` - apply 1-3 times.
 
-#### 2.2.1.3 "amount"
-**required for "radiation"** Radiation amount
-> Value: 0-100
+### "line"
 
-#### 2.2.1.4 "chance"
-**optional** one-in-??? chance to apply
-> Value: *number*
+| Field | Description
+|---    |---
+| "line" | (required) Set things in a line. Value: `"terrain", "furniture", "trap", "radiation"`. Requires "x", "y", "x2", "y2", "id" (or "amount" for "radiation"). Example: `{ "line": "terrain", "id": "t_lava", "x:" 5, "y": 5, "x2": 20, "y2": 20 }`
+| "id" | (required except by "radiation") Value: `"ter_id", "furn_id", or "trap_id"`. Example: `"id": "f_counter"`
+| "x", "y" | (both required) Start X, Y coordinates, either as a constant from `0-23`, or as a range `[ 0-23, 0-23 ]` for a random point in that range. Example: `"x": 12, "y": [ 5, 15 ]`
+| "x2", "y2" | (both required) End X, Y coordinates, either as a constant from `0-23`, or as a range `[ 0-23, 0-23 ]` for a random point in that range. Example: `"x": 22, "y": [ 15, 20 ]`
+| "amount" | (required for "radiation") Value: 0-100. Radiation amount
+| "chance" | (optional) One-in-N chance to apply
+| "repeat" | (optional) Value: `[ from_num, to_num ]`. Repeat this randomly between `from_num` and `to_num` times. Only makes sense if the coordinates are random. Example: `[ 1, 3 ]` - apply 1-3 times.
 
-#### 2.2.1.5 "repeat"
-**optional** repeat this randomly between ??? and ??? times. Only makes sense if the coordinates are random
-> Value: [ *number*, *number* ]
 
-Example: [ 1, 3 ] - apply 1-3 times
 
 ### 2.2.2 "square"
 **required** Define a square of things. Requires "x", "y", "x2", "y2", "id" (or "amount" for "radiation")
-> Value: "terrain", "furniture", "trap", "radiation"
+Value: `"terrain", "furniture", "trap", "radiation"`
+
+Example:
+```json
+{ "square": "radiation", "amount": 10, "x:" [ 0, 5 ], "y": [ 0, 5 ], "x2": [ 18, 23 ], "y2": [ 18, 23 ] }
```

Comment:
```suggestion
{ "square": "radiation", "amount": 10, "x": [ 0, 5 ], "y": [ 0, 5 ], "x2": [ 18, 23 ], "y2": [ 18, 23 ] }
```

Reply:
Thanks, yeah I noticed a few of those (in fact my vim markdown plugin highlighted them for me but I didn't see why at first). Resolved in the latest push.